### PR TITLE
Some APIs to help with out of core joins in Spark [skip ci]

### PR DESCRIPTION
**Pull request by revans2**
It adds a simple prefix sum for doing size calculations. It also changes a few APIs so that they take a ColumnView instead of a ColumnVector.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 1]
```diff
@@ -1303,6 +1303,17 @@ public final ColumnVector rollingWindow(Aggregation op, WindowOptions options) {
     }
   }
 
+  /**
+   * Compute the cumulative sum/prefix sum of the values in this column.
+   * This is similar to a rolling window SUM with unbounded preceding and none following.
+   * Input values 1, 2, 3
+   * Output values 1, 3, 6
+   * This currently only works for long values that are not nullable.
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:1311]:
```
This currently only works for long values...
```
Why?

File: java/src/main/native/src/ColumnViewJni.cpp [hunk 2]
```diff
@@ -1779,6 +1794,8 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_nansToNulls(JNIEnv *env,
   CATCH_STD(env, 0)
 }
 
+
+
```
Comment (jlowe) [inline java/src/main/native/src/ColumnViewJni.cpp:1798]: extraneous whitespace change?

File: java/src/main/native/src/prefix_sum.cu [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <thrust/scan.h>
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_device_view.cuh>
```
Comment (jlowe) [inline java/src/main/native/src/prefix_sum.cu:20]: Nit: I don't think this header is needed, as I don't see any use of a column device view.

**Review by jlowe — APPROVED** [review]
Minor nits but otherwise lgtm.

**Comment by revans2** [comment]
@gpucibot merge

**Pull request merged**